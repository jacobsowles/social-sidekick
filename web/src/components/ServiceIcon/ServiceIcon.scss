.service-icon {
  display: inline-block;

  p {
    margin-top: 5px;
    text-align: center;
  }
}
