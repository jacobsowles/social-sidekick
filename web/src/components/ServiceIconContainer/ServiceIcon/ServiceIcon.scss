@import '@styles/_variables.scss';

.service-icon {
  border: 2px dashed rgba(0, 0, 0, 0.1);
  display: inline-block;
  margin-right: 20px;
  padding: 5px;

  &:hover {
    border-color: $brand-primary;
    cursor: pointer;
  }

  &:last-child {
    margin-right: 0;
  }

  p {
    font-size: 0.8em;
    font-weight: bold;
    margin-top: 5px;
    text-align: center;
  }
}

.service-icon-connected {
  border: 2px solid rgba(0, 0, 0, 0.3);
  position: relative;

  &:hover {
    background: rgba(0, 0, 0, 0.1);
    border-color: rgba(0, 0, 0, 0.1);

    .badge {
      background: red;
      padding: 5px 7px;
    }
  }

  .badge {
    background: #207232;
    border-radius: 12px;
    box-shadow: 0 1px 2px 0 rgba(60, 64, 67, 0.3), 0 2px 6px 2px rgba(60, 64, 67, 0.15);
    padding: 5px;
    position: absolute;
    right: -8px;
    top: -8px;

    svg {
      color: #ffffff;
    }
  }
}

.service-icon-github svg {
  color: #333333;
}

.service-icon-twitter svg {
  color: #55acee;
}
