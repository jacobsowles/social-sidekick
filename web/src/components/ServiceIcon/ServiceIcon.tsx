import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

import './ServiceIcon.scss';

export interface ServiceIconProps {
  className?: string;
  iconName: string;
  isConnected: boolean;
  label?: string;
  size?:
    | 'xs'
    | 'sm'
    | 'lg'
    | '1x'
    | '2x'
    | '3x'
    | '4x'
    | '5x'
    | '6x'
    | '7x'
    | '8x'
    | '9x'
    | '10x'
    | undefined;
}

const ServiceIcon: FunctionComponent<ServiceIconProps> = ({
  children,
  className,
  iconName,
  isConnected,
  label,
  size,
  ...rest
}) => {
  const icons: IconProp[] = [
    { iconName: 'github', prefix: 'fab' },
    { iconName: 'twitter', prefix: 'fab' }
  ];
  let selectedIcon = icons.find(i => i.iconName === iconName);

  if (!selectedIcon) {
    selectedIcon = {
      iconName: 'question-circle',
      prefix: 'far'
    };
  }

  const classes = classNames(
    'service-icon',
    { 'service-icon-connected': isConnected },
    'service-icon-' + selectedIcon.iconName,
    className
  );

  return (
    <div className={classes}>
      {isConnected && (
        <span className="badge">
          <FontAwesomeIcon icon="check" />
        </span>
      )}
      <FontAwesomeIcon icon={[selectedIcon.prefix, selectedIcon.iconName]} size={size} {...rest} />
      {label && <p className="service-icon-label">{label}</p>}
    </div>
  );
};

ServiceIcon.defaultProps = {
  className: undefined,
  label: undefined,
  size: '4x'
};

interface IconProp {
  prefix: any;
  iconName: any;
}

export default ServiceIcon;
