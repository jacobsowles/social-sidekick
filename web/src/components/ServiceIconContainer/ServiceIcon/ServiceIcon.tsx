import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React, { FunctionComponent } from 'react';

import './ServiceIcon.scss';

export interface ServiceIconProps {
  badgeIcon?: IconProp;
  className?: string;
  icon: IconProp;
  label?: string;
  onClick: () => void;
  onMouseOut: () => void;
  onMouseOver: () => void;
}

const ServiceIcon: FunctionComponent<ServiceIconProps> = ({
  badgeIcon,
  className,
  icon,
  label,
  onClick,
  onMouseOut,
  onMouseOver
}) => {
  return (
    <div className={className} onClick={onClick} onMouseOut={onMouseOut} onMouseOver={onMouseOver}>
      {badgeIcon && (
        <span className="badge">
          <FontAwesomeIcon icon={badgeIcon} />
        </span>
      )}

      <FontAwesomeIcon icon={icon} size="4x" />

      {label && <p className="service-icon-label">{label}</p>}
    </div>
  );
};

export default ServiceIcon;
