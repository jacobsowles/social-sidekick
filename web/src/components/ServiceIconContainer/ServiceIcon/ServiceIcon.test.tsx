import React from 'react';

import { init, shallow, shallowTopLevelElement, toJson } from '@tests/component-test-base';
import ServiceIcon, { ServiceIconProps } from './ServiceIcon';

describe('ServiceIcon', () => {
  let props: ServiceIconProps;
  init(() => <ServiceIcon {...props} />, 'div');

  beforeEach(() => {
    props = {
      className: undefined,
      icon: 'github',
      onClick: jest.fn(),
      onMouseOut: jest.fn(),
      onMouseOver: jest.fn()
    };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });

  describe('when `className` is defined', () => {
    beforeEach(() => {
      props.className = 'another-class';
    });

    it('should include the specified class name', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.contain('another-class');
    });

    it('should include the service-icon class name', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.contain('service-icon');
    });
  });

  describe('when `className` is undefined', () => {
    it('should include the service-icon class name', () => {
      shallowTopLevelElement()
        .prop('className')
        .should.contain('service-icon');
    });
  });
});
