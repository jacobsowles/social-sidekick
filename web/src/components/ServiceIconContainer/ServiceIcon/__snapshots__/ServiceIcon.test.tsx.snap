// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceIcon should render the component without crashing 1`] = `
<div
  className="service-icon"
  onClick={[MockFunction]}
  onMouseOut={[MockFunction]}
  onMouseOver={[MockFunction]}
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="github"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size="4x"
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</div>
`;
